event-sourcing/event-subscription-registry/src/main/java/uk/gov/justice/subscription/registry/EventSourceDefinitionRegistry.java

```diff
@@ -1,39 +1,60 @@
 package uk.gov.justice.subscription.registry;
 
-import static java.util.Optional.ofNullable;
-import static java.util.stream.Collectors.toMap;
-
 import uk.gov.justice.subscription.domain.eventsource.EventSourceDefinition;
 
+import java.util.HashMap;
 import java.util.Map;
-import java.util.Optional;
-import java.util.function.BinaryOperator;
-import java.util.stream.Stream;
+
+import javax.enterprise.inject.Vetoed;
 
 /**
  * Registry containing {@link EventSourceDefinition}s mapped by the event source name
  */
+@Vetoed
 public class EventSourceDefinitionRegistry {
 
-    private final Map<String, EventSourceDefinition> registry;
+    private final Map<String, EventSourceDefinition> registry = new HashMap<>();
 
-    private final BinaryOperator<EventSourceDefinition> registerFirstEventSourceWhenDuplicate =
-            (eventSource, eventSource2) -> eventSource;
-
-    public EventSourceDefinitionRegistry(final Stream<EventSourceDefinition> eventSources) {
-        registry = eventSources.collect(toMap(
-                EventSourceDefinition::getName,
-                eventSourceDefinition -> eventSourceDefinition,
-                registerFirstEventSourceWhenDuplicate));
-    }
+    private EventSourceDefinition defaultEventSourceDefinition;
 
     /**
-     * Return an {@link EventSourceDefinition} mapped to an event source name or empty if not mapped.
+     * Return an {@link EventSourceDefinition} mapped to an event source name or empty if not
+     * mapped.
      *
      * @param eventSourceName the event source name to look up
      * @return Optional of {@link EventSourceDefinition} or empty
      */
-    public Optional<EventSourceDefinition> getEventSourceDefinitionFor(final String eventSourceName) {
-        return ofNullable(registry.get(eventSourceName));
+    public EventSourceDefinition getEventSourceDefinitionFor(final String eventSourceName) {
+        return registry.get(eventSourceName);
+    }
+
+    public EventSourceDefinition getDefaultEventSourceDefinition() {
+        if (defaultEventSourceDefinition == null) {
+            throw new RegistryException("You must define a default event source in event-sources.yaml");
+        }
+        return defaultEventSourceDefinition;
+    }
+
+    public void register(final EventSourceDefinition eventSourceDefinition) {
+        if (eventSourceDefinition.isDefaultEventSource()) {
+            validateDefaultEventSource(eventSourceDefinition);
+            defaultEventSourceDefinition = eventSourceDefinition;
```

Comment:
We need to store the default definition in the map, otherwise it cannot be looked up by name